DEVELOPMENT.md

```diff
@@ -0,0 +1,71 @@
+
+The Datadog Provider can be built to use the binary as a terraform plugin. This is most useful when attempting to build off a feature branch and manually run the `terraform plan|apply` commands on a real HCL configuration.
+
+This provider can be built by running `make build`, or just `make`. This will place the binary in `$GOPATH/bin`
+
+Running `terraform init` in a folder containing `.tf` files will let terraform look for any needed providers. The default behavior is to pull the latest released binary, but you can tell the terraform CLI to look for your locally built binary instead.
+
+The steps for this approach can differ depending on the version of Terraform being used. More information can be found on the official [Terraform documentation][4].
+
+## Testing the Provider
+
+The Datadog terraform provider uses the standard [Terraform Testing Framework][6] to define its tests.
+
+**NOTE** Use the API and APP keys for a sandbox/test organization, never an account hosting production data. This test suite will create/update/delete real resources.
+
+-   `DD_API_KEY="<api_key>" DD_APP_KEY=<app_key> make testacc` will run the test suite against the real Datadog API.
+
+We also use `cassettes` to record API request/responses, which allows the test suite to be reliable and run very quickly. There are a few environment variables that can control this behavior (All commands are assumed to be prefixed with `DD_API_KEY=` and `DD_APP_KEY=`)
+
+-   `TESTARGS`: Allows passing extra flags directly to the underlying `go test` command. Most often used to run individual tests:
+    -   Ex: `TESTARGS="-run TestAccDatadogServiceLevelObjective_Basic" make test`
+-   `RECORD`: This denotes whether the test suite should interact with the real Datadog API. Available options are:
+    -   `RECORD=none make testacc`: Run against the real Datadog API.
+    -   `RECORD=true make testacc`: Run against the real Datadog API and record the request/response to be used later.
+    -   `RECORD=false make testacc`: Don't interact with the real Datadog API, instead playback against the recorded cassettes.
+
+## Generating Documentation
+
+Documentation for this provider is autogenerated using the [`tfplugindocs`][8] CLI.
+
+-   Ensure each Schema attribute in the code contains a `Description` field
+-   Place example HCL configuration files or import scripts in the `examples` folder under your resource/data-source
+-   Run `tfplugindocs` to autogenerate updates to the `docs` folder!
+
+## Updating the underlying Datadog SDK Clients
+
+In order to update the underlying API Clients that are used by this provider to interact with the Datadog API, run:
+
+```sh
+API_CLIENT_VERSION=vx.y.z ZORKIAN_VERSION=vx.y.z make update-go-client
+```
+
+where:
+
+-   `API_CLIENT_VERSION` is the version or commit ref of the https://github.com/DataDog/datadog-api-client-go client.
+-   `ZORKIAN_VERSION` is the version or commit ref of the https://github.com/zorkian/go-datadog-api client.
```

Comment:
I'll add a note like "if you run it just after a release, this will automatically pick up the latest tag without needing to specify the version".